tools/depmod.c

```diff
@@ -2343,12 +2344,12 @@ static int output_symbols_bin(struct depmod *depmod, FILE *out)
 			ret = -ENOMEM;
 			goto err_scratchbuf;
 		}
-		memcpy(scratchbuf_str(&salias) + baselen, sym->name, len + 1);
-		duplicate =
-			index_insert(idx, alias, sym->owner->modname, sym->owner->idx);
+		s = scratchbuf_str(&salias);
```

Comment:
ugh... good find. As a follow up commit, can you rename alias to something else like `aliasbuf`  so it doesn't get misused?

Reply:
Adjusted. While at it, inititalized the buffer directly without the need to use memcpy.